selftests/functional/test_task_timeout.py

```diff
@@ -0,0 +1,38 @@
+from avocado.core.job import Job
+from avocado.utils import script
+from avocado.utils.network.ports import find_free_port
+
+from .. import TestCaseTmpDir, skipUnlessPathExists
+
+SCRIPT_CONTENT = """#!/bin/bash
+/bin/sleep 30
+"""
+
+
+class TaskTimeOutTest(TestCaseTmpDir):
+
+    def setUp(self):
+        super(TaskTimeOutTest, self).setUp()
+        self.script = script.TemporaryScript(
+            'sleep.sh',
+            SCRIPT_CONTENT,
+            'avocado_timeout_functional')
+        self.script.save()
+
+    @skipUnlessPathExists('/bin/sleep')
+    def test_sleep_longer_timeout(self):
+        status_server = '127.0.0.1:%u' % find_free_port()
+        config = {'run.references': [self.script.path],
+                  'nrunner.status_server_listen': status_server,
+                  'nrunner.status_server_uri': status_server,
+                  'task.timeout.running': 2,
+                  'run.test_runner': 'nrunner'}
+
+        with Job.from_config(job_config=config) as job:
+            job.run()
+            self.assertEqual(1, job.result.skipped)
```

Comment:
Nitpick: better move the assertions to out of the job context manager.  Gives better readability and confidence that the job is finished.